src/debug-generator/debug-generator.c

```diff
@@ -59,12 +62,26 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
                 if (value) {
                         r = parse_boolean(value);
-                        if (r < 0)
-                                log_error("Failed to parse systemd.debug_shell= argument '%s', ignoring.", value);
-                        else
-                                arg_debug_shell = r;
-                } else
-                        arg_debug_shell = true;
+                        if (r > 0) {
+                                arg_debug_shell = strdup(DEBUGTTY);
+                                if (!arg_debug_shell)
+                                        return -ENOMEM;
+                        } else if (r < 0) {
+                                const char *tty;
+
+                                tty = skip_dev_prefix(value);
+                                if (!startswith(tty, "tty"))
+                                        return log_error("Failed to set debug shell argument '%s', ignoring.", value);
```

Comment:
error is misleading. should say that the specified device doesn't begin with tty

Reply:
that said, I figure the check should go away entirely, there are valid ttys that do not begin with "tty" in the name. quoting our own udev rules files:

```udev-rules
SUBSYSTEM=="tty", KERNEL=="tty[a-zA-Z]*|hvc*|xvc*|hvsi*|ttysclp*|sclp_line*|3270/tty[0-9]*", TAG+="systemd"
```
all those devices are valid ttys at least we know of, hence I'd just drop the extra check. it's a debugging tool anyway, the user is hopefully smart enough to specify something sensible...